### PR TITLE
LibPDF: Start implementing separation color spaces

### PR DESCRIPTION
**Pull request by nico**
When printing, a separation color space is the name of one ink and an amount.

When drawing to a screen, that amount gets converted by a PDF function to inputs of an alternate color space.

This PR adds the separation color space bits for drawing to a screen, and starts implementing PDF functions. For now, it implements one of four possible function types (the one used in 66% of separation color spaces, in my test set).

### CODE REVIEW

**Comment by nico** [comment]
Before:

Screenshot 2023-11-05 at 10 39 19 PM

After:

Screenshot 2023-11-05 at 10 39 04 PM

File: Userland/Libraries/LibPDF/ColorSpace.cpp [hunk 1]
```diff
@@ -687,16 +687,70 @@ Vector<float> IndexedColorSpace::default_decode() const
...
+        return Error { Error::Type::MalformedPDF, "Separation color space expected three parameters" };
+
+    // "The name parameter is a name object specifying the name of the colorant that this Separation color space
+    //  is intended to represent (or one of the special names All or None; see below)"
+    auto param0 = TRY(document->resolve(parameters[0]));
+    if (!param0.has<NonnullRefPtr<Object>>())
+        return Error { Error::Type::MalformedPDF, "Separation color space expects object for first arg" };
+    auto name_object = param0.get<NonnullRefPtr<Object>>();
+    if (!name_object->is<NameObject>())
+        return Error { Error::Type::MalformedPDF, "Separation color space expects name object for first arg" };
```
Comment (awesomekling) [inline Userland/Libraries/LibPDF/ColorSpace.cpp:702]: Would it make sense to use a `TRY` pattern here?
Something like
```cpp
auto name_object = TRY(param0.get_if<NonnullRefPtr<Object>>());
```

Comment (awesomekling) [reply Userland/Libraries/LibPDF/ColorSpace.cpp:702]: Maybe even some kinda
```cpp
auto name_object = TRY(param0.get_if<NameObject>());
```

Comment (nico) [reply Userland/Libraries/LibPDF/ColorSpace.cpp:702]: Belated: #21921

**Pull request merged**